pandas/core/arrays/interval.py

```diff
@@ -1026,6 +1036,58 @@ def __array__(self, dtype=None):
                 result[i] = Interval(left[i], right[i], closed)
         return result
 
+    def __arrow_array__(self, type=None):
+        """
+        Convert myself into a pyarrow Array.
+        """
+        import pyarrow as pa
+
+        try:
+            subtype = pa.from_numpy_dtype(self.dtype.subtype)
+        except TypeError:
+            raise TypeError(
+                "Conversion to arrow with subtype '{}' "
+                "is not supported".format(self.dtype.subtype)
+            )
+        interval_type = ArrowIntervalType(subtype, self.closed)
+        storage_array = pa.StructArray.from_arrays(
+            [
+                pa.array(self.left, type=subtype, from_pandas=True),
+                pa.array(self.right, type=subtype, from_pandas=True),
+            ],
+            names=["left", "right"],
+        )
```

Comment:
ideally would share this code as much as possible in a base mixin , might require spitting  some function / methods

Reply:
ok as a follow up but if possible now pls do it

Reply:
This whole function here is rather interval specific (eg the fact that the storage array is a StructArray, the fact that we need the handle the mask in a special way, the conversion of the subtype, .. are all specific to intervals)